src/worker.cpp

```diff
@@ -1563,4 +1563,100 @@ void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
         */
     }
 }
+
+void Worker::deleteFruVpd(const std::string& i_dbusObjPath)
+{
+    if (i_dbusObjPath.empty())
+    {
+        throw std::runtime_error("Given DBus object path is empty.");
+    }
+
+    const std::string& l_fruPath =
+        jsonUtility::getFruPathFromJson(m_parsedJson, i_dbusObjPath);
+
+    try
+    {
+        const bool l_presentPropValue =
+            std::get<bool>(dbusUtility::readDbusProperty(
+                constants::pimServiceName, i_dbusObjPath,
+                constants::inventoryItemInf, "Present"));
+
+        if (!l_presentPropValue)
+        {
+            throw std::runtime_error("Given FRU is not present");
+        }
+        else
+        {
+            if (jsonUtility::isActionRequired(m_parsedJson, l_fruPath,
+                                              "preAction", "deletion"))
+            {
+                if (!jsonUtility::executeBaseAction(m_parsedJson, "preAction",
+                                                    l_fruPath, "deletion"))
+                {
+                    throw std::runtime_error("Pre action failed");
+                }
+            }
+
+            std::vector<std::string> l_interfaceList{
+                constants::operationalStatusInf};
+
+            types::MapperGetSubTree l_subTreeMap =
+                dbusUtility::getObjectSubTree(i_dbusObjPath, 0,
+                                              l_interfaceList);
+
+            types::ObjectMap l_objectMap;
+
+            // Updates VPD specific interfaces property value under PIM for sub
+            // FRUs.
+            for (const auto& [l_objectPath, l_serviceInterfaceMap] :
+                 l_subTreeMap)
+            {
+                types::InterfaceMap l_interfaceMap;
+                vpdSpecificUtility::resetDataUnderPIM(l_objectPath,
+                                                      l_interfaceMap);
+                l_objectMap.emplace(l_objectPath, std::move(l_interfaceMap));
+            }
+
+            types::InterfaceMap l_interfaceMap;
+            vpdSpecificUtility::resetDataUnderPIM(i_dbusObjPath,
+                                                  l_interfaceMap);
+
+            l_objectMap.emplace(i_dbusObjPath, std::move(l_interfaceMap));
+
+            if (!dbusUtility::callPIM(std::move(l_objectMap)))
+            {
+                throw std::runtime_error("Call to PIM failed");
+            }
+
+            if (jsonUtility::isActionRequired(m_parsedJson, l_fruPath,
+                                              "postAction", "deletion"))
+            {
+                if (!jsonUtility::executeBaseAction(m_parsedJson, "postAction",
```

Comment:
why worker functions for processPostAction not used ?

Reply:
Same here.

Reply:
Done